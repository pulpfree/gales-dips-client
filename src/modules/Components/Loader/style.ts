
import { Theme, makeStyles, createStyles } from '@material-ui/core/styles'

export const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    container: {
      position: 'relative',
      // left: theme.spacing.unit * 10,
    },
    progress: {
      margin: theme.spacing(2),
    },
  })
)