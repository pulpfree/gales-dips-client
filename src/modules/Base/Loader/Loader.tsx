import React from 'react'

import CircularProgress from '@material-ui/core/CircularProgress'

import { useStyles } from './style'

export const Loader = (): JSX.Element => {
  const classes = useStyles()

  return (
    <div className={classes.container} data-testid='loader'>
      <CircularProgress className={classes.progress} />
    </div>
  )
}
