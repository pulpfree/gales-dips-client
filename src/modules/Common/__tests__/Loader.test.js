/* eslint no-undef: "off" */

import React from 'react'
import { shallow } from 'enzyme'

import Loader from '../Loader'

describe('Loader', () => {

  let mountedC
  let props
  const reportC = () => {
    if (!mountedC) {
      mountedC = shallow(
        <Loader {...props} />
      )
    }
    return mountedC
  }

  beforeEach(() => {
    props = {
    }
    mountedC = undefined
  })

  it('matches snapshot', () => {
    expect(reportC()).toMatchSnapshot()
  })

})