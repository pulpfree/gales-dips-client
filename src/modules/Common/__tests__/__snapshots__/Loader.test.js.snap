// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Loader) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "container": "Loader-container-1",
        "progress": "Loader-progress-2",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "container": "Loader-container-1",
          "progress": "Loader-progress-2",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
